homeassistant/components/sfr_box/diagnostics.py

```diff
@@ -26,13 +26,17 @@ async def async_get_config_entry_diagnostics(
             "data": dict(entry.data),
         },
         "data": {
-            "dsl": async_redact_data(dataclasses.asdict(data.dsl.data), TO_REDACT),
+            "dsl": async_redact_data(
+                dataclasses.asdict(await data.system.box.dsl_get_info()), TO_REDACT
+            ),
```

Comment:
When infrastructure is FTTH, the DSL coordinator is never initialised which resulted in `dataclasses.asdict` failing.
Now we get the latest available data.